hashes/src/internal_macros.rs

```diff
@@ -188,6 +180,18 @@ macro_rules! hash_type {
             fn internal_new(arr: [u8; $bits / 8]) -> Self { Hash(arr) }
 
             fn internal_engine() -> HashEngine { Default::default() }
+
+            /// Constructs a hash from the underlying byte array.
+            pub const fn from_byte_array(bytes: <Self as crate::Hash>::Bytes) -> Self {
+                Hash(bytes)
+            }
+
+            /// Returns an all zero hash.
+            ///
+            /// An all zeros hash is a made up construct because there is not a known input that can create
+            /// it, however it is used in various places in Bitcoin e.g., the Bitcoin genesis block's
```

Comment:
Grammar nit: change `it, however` to `it. However, `. (comma to period, then add a comma after howenver).

I know the typo was in the original comment, but let's fix it now.

Reply:
You will need to change this in several other places as well.